packages/aws-cdk-lib/aws-events/lib/archive.ts

```diff
@@ -32,6 +33,16 @@ export interface BaseArchiveProps {
    * @default - Infinite
    */
   readonly retention?: Duration;
+
+  /**
+   * The customer managed key that encrypts this archive
+   * Can be either a string that represents the Key ARN directly or a kms.IKey
+   * This can potentially be a string because if it is an empty string,
+   * and it is used to update the archive, then Archive reverts back to using an AWS owned key
+   *
+   * @default - Use an AWS managed key
+   */
+  readonly kmsKey?: kms.IKey | string;
```

Comment:
We don't allow type unions at the L2 level. This should be just `kms.IKey`.

Reply:
Is this completely not possible? Because for the constructed template, we have this specific case that requires this.  

If we create a CFN Stack with a CMK, then later tries to remove it, we need to do the following:
1. Create a stack with CMK
2. Remove the KMS by **setting the KMS Key in the template to an empty string**
3. Update the stack

Is it possible that we supply a key that has the keyArn as an empty string? Or do we need to modify our L1 construct definitions for this?

Reply:
> Is this completely not possible

It is possible, but we don't do it because of poor ergonomics. ([Design Guidelines](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md#unions))

It is not a design goal of the L2 library to be out-of-the-box compatible with whatever L1 setup you happen to have configured inside your team, sorry. If you need to maintain compatibility in an existing template, you can always use [escape hatches](https://docs.aws.amazon.com/cdk/v2/guide/cfn-layer.html).